src/pages/FilesPage/FilesPage.tsx

```diff
@@ -1,55 +1,58 @@
 import { useMutation, useQuery, useQueryClient } from 'react-query';
 import Container from '@mui/material/Container';
-import Divider from '@mui/material/Divider';
 import Grid from '@mui/material/Grid';
 import LinearProgress from '@mui/material/LinearProgress';
 
-import AddIcon from '@mui/icons-material/Add';
-import DeleteIcon from '@mui/icons-material/Delete';
-import DownloadIcon from '@mui/icons-material/Download';
-
-import {
-	FileMenuButton,
-	FilesContainer,
-	FilesGridContainer,
-	FlexBox,
-	UHeader,
-} from './FilePage.styles';
+import { FilesContainer, FilesGridContainer } from './FilePage.styles';
+import theme from 'themes/theme';
 
 // helpers
 import { FileItem } from './components/FileItem';
-import { FileLinkResponse, FileService } from 'clients/CoreService';
+import { FileManagerService, PostUrlInfo } from 'clients/CoreService';
 import { useSnackbarOnError } from 'hooks/notification/useSnackbarOnError';
 import { entities } from 'consts/entities';
 import { FileInfo } from './components/FileItem/FileItem';
+import { FolderItem } from './components/FolderItem';
+import { ControlMenu } from './components/ControlMenu';
+import { LocationLinks } from './components/LocationLinks';
+import { uploadFileToS3 } from 'shared/fileUploadUtil';
+import { downloadFilesByLink } from './components/utils/downloadFilesByLink';
+import { getBasename } from 'shared/util';
 
 function FilesPage() {
-	const coreUrl: string = process.env.REACT_APP_CORE_URL || '';
 	const queryClient = useQueryClient();
+	const [currentLocation, setCurrentLocation] = useState<string>('');
 	const [processingFiles, setProcessingFiles] = useState<Array<FileInfo>>([]);
-	const [checkedIds, setCheckedIds] = useState<Array<number>>([]);
+	const [checkedItems, setCheckedItems] = useState<Array<string>>([]);
 
-	const { data: existingFiles, isLoading: isFilesLoading } = useQuery(
-		[entities.file],
-		() => FileService.list(),
+	const { data: existingContent, isLoading: isContentLoading } = useQuery(
+		[entities.file, currentLocation],
+		() => FileManagerService.list(currentLocation),
 		{
 			onError: useSnackbarOnError(),
 		},
 	);
 
 	const { mutate: uploadFiles } = useMutation(
 		[entities.file],
-		(files: Array<File>) => {
-			return Promise.all(
-				files.map(async (file: File) => {
-					await FileService.upload({ file });
+		async (files: Array<File>) => {
+			const { urls } = await FileManagerService.getSignedPostUrls({
+				location: currentLocation,
+				fileInfos: files.map(({ name, size }) => ({ name, size })),
+			});
 
-					await queryClient.invalidateQueries(entities.file);
-					setProcessingFiles(processingFiles =>
-						processingFiles?.filter(processedFile => processedFile.name !== file.name),
-					);
+			return Promise.all(
+				urls.map((postUrl: PostUrlInfo) => {
+					const fileName = getBasename(postUrl.path);
+					const file = files.find(file => file.name === fileName);
+					if (!file) throw new Error('Bad link generation');
```

Comment:
create variable `onError = useSnackbarOnError()` upper and do `onError(new Error('Bad link generation'))` here